server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1088,8 +1188,75 @@ protected void parseCreateField(ParseContext context) throws IOException {
         }
     }
 
+    @Override
+    public CheckedConsumer<IndexTimeScriptContext, IOException> getPostParsePhase() {
+        if (this.script == null) {
+            return null;
+        }
+        return c -> this.script.compiledScript.executeAndIndex(
+            c.searchLookup,
+            c.leafReaderContext,
+            0,
+            v -> this.indexValue(c.pc, v)
+        );
+    }
+
     @Override
     public FieldMapper.Builder getMergeBuilder() {
         return new Builder(simpleName(), type, ignoreMalformedByDefault, coerceByDefault).init(this);
     }
+
+    private interface NumberMapperScript {
+        void executeAndIndex(SearchLookup lookup, LeafReaderContext ctx, int doc, Consumer<Number> emitter);
+    }
+
+    private static class LongMapperScript implements NumberMapperScript {
+
+        final LongFieldScript.Factory scriptFactory;
+        final Map<String, Object> scriptParams;
+        final String fieldName;
+
+        private LongMapperScript(String fieldName, LongFieldScript.Factory scriptFactory, Map<String, Object> scriptParams) {
+            this.scriptFactory = scriptFactory;
+            this.scriptParams = scriptParams;
+            this.fieldName = fieldName;
+        }
+
+        @Override
+        public void executeAndIndex(SearchLookup lookup, LeafReaderContext ctx, int doc, Consumer<Number> emitter) {
+            LongFieldScript s = scriptFactory
```

Comment:
Do you think it'd be cool if `LongFieldScript` took a `LongConsumer`?

Reply:
As in, `runFocDoc(doc, consumer)` type of thing?

Reply:
It'd be reasonable to save that for a follow up. I'm convinced it'd save code and wouldn't slow things down but I'm not convinced it'd be a hell of a lot cleaner. It's not a blocker at least.